api/rds.proto

```diff
@@ -137,6 +137,16 @@ message RouteAction {
     WeightedCluster weighted_clusters = 3;
   }
 
+  enum ClusterNotFoundResponseCode {
+    // HTTP status code - 503 Service Unavailable.
+    SERVICE_UNAVAILABLE = 0;
+    // HTTP status code - 404 Not Found.
+    NOT_FOUND = 1;
```

Comment:
Why don't we just make this a `uint16`? Maybe folks will want to pick any kind of 4xx or 5xx variant?

Reply:
OTOH, we also have https://github.com/envoyproxy/data-plane-api/blob/master/api/rds.proto#L304 which is a reponse code enum. It really depends on how customizable we want this to be ¯\_(ツ)_/¯

Reply:
+1 on just making it a uint16.  If someone wants to configure to do something odd, I don't think there's a good reason to prevent that.  Mabye they want 500/502 instead of 503?  Or for some reason they know this condition means 401 (unauthorized) in their environment?

Reply:
uint16 for this case is probably fine, though I would constrain it to be >= 200 and <= 599.

Reply:
Yep; this would be a perfect use of the validate.proto annotations.

Reply:
I'm going to push back on this.

1. 99% of HTTP status codes in 200-599 range should not be returned for this particular error (i.e. when route is configured to use cluster that's not configured). The meaning of each HTTP status code is clearly defined and returning arbitrary HTTP status code is just wrong.

2. This would be inconsistent at the API level with [`RedirectAction`](https://github.com/envoyproxy/data-plane-api/blob/ec38af2de5f1c218dfb59a5592fe007ad7fee785/api/rds.proto#L401), which is the only other configureable HTTP status code to return (AFAIK).

3. Virtually everywhere else in the code HTTP status code to return is hardcoded and not configurable at all, yet for this particular error we want to allow anything?

Reply:
I don't really care one way or the other. I will defer to @htuch and @ggreenway.

Reply:
I agree that most of the codes are wrong; there's a few that could be correct, such as a raw 500.  However, I'm ok with leaving it how it is, and if someone else needs another strange code they can make a PR.